src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSPackageMetadataTransformer.java

```diff
@@ -45,19 +45,29 @@
     NodeJSPackageMetadataNamer namer =
         new NodeJSPackageMetadataNamer(
             apiConfig.getPackageName(), apiConfig.getDomainLayerLocation());
-    models.add(generateMetadataView(namer));
+    models.add(generateMetadataView(model, apiConfig, namer));
     return models;
   }
 
-  private ViewModel generateMetadataView(NodeJSPackageMetadataNamer namer) {
+  private ViewModel generateMetadataView(
+      Model model, ApiConfig apiConfig, NodeJSPackageMetadataNamer namer) {
     return PackageMetadataView.newBuilder()
         .templateFileName(PACKAGE_FILE)
         .outputPath("package.json")
         .identifier(namer.getMetadataIdentifier())
-        .version(PACKAGE_VERSION)
-        .gaxVersion(GAX_VERSION)
-        .protoVersion(PROTO_VERSION)
-        .url(PACKAGE_URL)
+        .packageVersion(packageConfig.packageVersion("nodejs"))
+        .protoPath(packageConfig.protoPath())
+        .shortName(packageConfig.shortName())
+        .gaxVersion(packageConfig.gaxVersion("nodejs"))
+        .protoVersion(packageConfig.protoVersion("nodejs"))
+        .commonProtosVersion(packageConfig.commonProtosVersion("nodejs"))
+        .packageName(packageConfig.packageName("nodejs"))
```

Comment:
What would you suggest here? Is it sufficient that this to be changed to an enum, or are you specifically concerned with the repetition? (I think it's probably overkill to create a per-language "view" of the config object.)